app/scripts/common-transforms.js

```diff
@@ -174,6 +192,123 @@ var commonTransforms = (function(_) {
             });
 
             return arrayToReturn;
+        },
+
+        getClickableObjectArr: function (records, type) {
+            var result = [];
+            if(records) {
+                if(records.constructor === Array) {
+                    records.forEach(function(record) {
+                        if(record.name) {
+                            var obj = {
+                                _id: record.uri,
+                                _type: type,
+                                title: record.name,
+                                subtitle: ''
+                            };
+                            result.push(obj);
+                        }
+                    });
+                } else {
+                    var obj = {
+                        _id: records.uri,
+                        _type: type,
+                        title: records.name,
+                        subtitle: ''
+                    };
+                    result.push(obj);
+                }
+            }
+            
+            return result;
+        },
+
+        combineArrays: function(arr1, arr2) {
```

Comment:
I believe you should be able to just use _.concat instead of writing your own function.  https://lodash.com/docs#concat